File: recipe-management-domain/src/main/java/fr/cta/recipe/management/domain/usecase/CreateRecipeUseCase.java
package fr.cta.recipe.management.domain.usecase;

import fr.cta.recipe.management.domain.valueobject.QuantifiedIngredient;
import lombok.Value;

import java.util.List;

public class CreateRecipeUseCase implements UseCase<CreateRecipeUseCase.CreateRecipeInput, CreateRecipeUseCase.CreateRecipeOutput> {

    @Override
    public CreateRecipeOutput execute(CreateRecipeInput input) {
        return null;
    }

    @Value
    public static class CreateRecipeInput {
        List<QuantifiedIngredient> quantifiedIngredients;
        List<String> steps;
    }

    public static class CreateRecipeOutput {
    }
}

File: recipe-management-domain/src/main/kotlin/fr/cta/recipe/management/domain/usecase/CreateRecipeUseCase.kt
package fr.cta.recipe.management.domain.usecase

import fr.cta.recipe.management.domain.entity.Recipe
import fr.cta.recipe.management.domain.valueobject.QuantifiedIngredient
import org.mapstruct.Mapper

class CreateRecipeUseCase: UseCase<CreateRecipeUseCase.CreateRecipeRequest, Recipe> {
    class CreateRecipeRequest(
            val quantifiedIngredients: List<QuantifiedIngredient>,
            val steps: List<String>,
        ) {
    }

    override fun execute(useCaseData: CreateRecipeRequest): Recipe {
        TODO("Not yet implemented")
    }
}